token-lending/program/src/processor.rs

```diff
@@ -1291,6 +1292,7 @@ fn process_repay_obligation_liquidity(
         msg!("Obligation lending market does not match the lending market provided");
         return Err(LendingError::InvalidAccountInput.into());
     }
+    // @FIXME: why is it necessary to refresh before repay?
```

Comment:
I spotted these during review. I'll follow up in another PR.

Reply:
The borrow balance includes interest so when you repay you need to ensure that interest has been accrued first

Reply:
Yeah, you're right. I was thinking about the fact that the withdraw collateral instruction is separate now, and that requires refreshing all the reserves and obligation, so it might not be needed on repay. However, if you were able to repay without accruing interest, refresh, and then withdraw, the interest wouldn't accrue correctly in the refresh because your balance had been changed.

Reply:
Removed these comments.

Reply:
Great, another thing is that I remember in the initial multi-collateral PR of yours we discussed how it would be ideal you could do multiple obligation operations in the same transaction, not sure if the stale flag on obligations would prevent any desired lending flows at this point.